containers/ddev-php-base/Dockerfile

```diff
@@ -31,30 +31,25 @@ RUN apt-get -qq install --no-install-recommends --no-install-suggests -y \
     wget
 RUN url="https://github.com/mikefarah/yq/releases/latest/download/yq_linux_${TARGETPLATFORM#linux/}"; wget ${url} -q -O /usr/bin/yq && chmod +x /usr/bin/yq
 ADD generic-files /
+
+RUN curl https://nginx.org/keys/nginx_signing.key | gpg --dearmor \
+    | tee /usr/share/keyrings/nginx-archive-keyring.gpg >/dev/null
+RUN echo "deb [signed-by=/usr/share/keyrings/nginx-archive-keyring.gpg] \
+http://nginx.org/packages/debian `lsb_release -cs` nginx" > /etc/apt/sources.list.d/nginx.list
+
+RUN curl -sSLo /tmp/debsuryorg-archive-keyring.deb https://packages.sury.org/debsuryorg-archive-keyring.deb && \
+    dpkg -i /tmp/debsuryorg-archive-keyring.deb && rm -f /tmp/debsuryorg-archive-keyring.deb && \
+    echo "deb [signed-by=/usr/share/keyrings/deb.sury.org-php.gpg] https://packages.sury.org/php/ $(lsb_release -sc) main" > /etc/apt/sources.list.d/php.list && apt-get -qq update
 #END base
 
-### ---------------------------ddev-xdebug-build--------------------------------------
+### ---------------------------ddev-php-extension-build--------------------------------------
+FROM base AS ddev-php-extension-build
+SHELL ["/bin/bash", "-c"]
 ### Xdebug compile specific version because 3.3 is not fully reliable
 ### See https://github.com/ddev/ddev/issues/6159
-### We don't need to recompile every Xdebug library https://xdebug.org/docs/compat (only PHP 8.0, 8.1, 8.2 can have Xdebug 3.2)
-### PECL does not allow you to install multiple versions of Xdebug, so there is `rm -f xdebug.reg`
-FROM base AS ddev-xdebug-build
-SHELL ["/bin/bash", "-c"]
-RUN curl -sSLo /tmp/debsuryorg-archive-keyring.deb https://packages.sury.org/debsuryorg-archive-keyring.deb && \
-    dpkg -i /tmp/debsuryorg-archive-keyring.deb && rm -f /tmp/debsuryorg-archive-keyring.deb && \
-    echo "deb [signed-by=/usr/share/keyrings/deb.sury.org-php.gpg] https://packages.sury.org/php/ $(lsb_release -sc) main" > /etc/apt/sources.list.d/php.list && apt-get update
-ARG XDEBUG_VERSION="3.2.2"
-ARG XDEBUG_BUILD_PACKAGES="build-essential php-pear php8.0-dev php8.1-dev php8.2-dev"
-RUN set -eu -o pipefail; \
-    apt-get -qq update && \
-    apt-get -qq install --no-install-recommends --no-install-suggests -y ${XDEBUG_BUILD_PACKAGES}
-RUN pecl channel-update pecl.php.net && \
-    for version in 8.0 8.1 8.2; do \
-        (apt-get -qq remove -y php${version}-xdebug || true) && \
-        pecl -d php_suffix=${version} install -f xdebug-${XDEBUG_VERSION} && \
-        rm -f /usr/share/php/.registry/.channel.pecl.php.net/xdebug.reg || exit $?; \
-    done
-#END ddev-xdebug-build
+RUN /usr/local/bin/build_php_extension.sh "php8.1" "xdebug" "3.2.2" "/usr/lib/php/20210902/xdebug.so"
+RUN /usr/local/bin/build_php_extension.sh "php8.2" "xdebug" "3.2.2" "/usr/lib/php/20220829/xdebug.so"
+#END ddev-php-extension-build
```

Comment:
We should be able to add our own builds here for PHP extensions that are not available in deb.sury.org repository.

Reply:
Great stuff, thanks for refactoring this.